### ISSUE TITLE
LEDs stay on when shutting down the PC

### ISSUE THREAD

**Issue by Infidaelity** [psieg/Lightpack#206]
Hi all,

so I've built myself an ambilight using this tutorial: https://www.youtube.com/watch?v=juC-3imLoTQ

Initially it worked fine and the LEDs always turned off when shutting down my PC.
But due to the windows feature "Fast boot" my PC only really went to standby instead of completely shutting down.
This also produced some other issues on my PC unrelated to lightpack which is why I turned this feature off two days ago.

Now since my PC is finally really shutting down when i want it to the LEDs stay turned on after shutting down the PC. At least most of the time. I had one exception so far where they turned off properly.

I already checked and made sure that all 3 settings in Prismatik to keep the LEDs on in various states (lock, standby, exit) are turned off.

Did any of you experience this and have an idea how to fix it so the LEDs properly shut down when I turn off my PC?

Looking forward to your answer!
Infidaelity

**Comment by psieg** [comment]
The fast boot setting should not have an effect on this.
Do you have DX hooking on?

**Comment by Infidaelity** [comment]
DX hookkng is currently disabled.
Should I enable it?

Just yesterday it happened again without even playing a game, just regular use of my PC (watching videos, browsing the web).
You could turn the LEDs off just fine manually in Prismatik but when I shut down the PC the LEDs stayed on.

**Comment by psieg** [comment]
No keep it off. Do the lights turn off when you lock/standby/exit?

**Comment by Infidaelity** [comment]
So far they turned off fine when locking.
I rarely use standby but on my few occasions they also turned off fine.
Exiting Prismatik doesn't turn the LEDs off although although all the "Keep lights ON after ..." settings are disabled.

**Comment by psieg** [comment]
Hmmm, it should. Can you turn up the log level and post a logfile please?

**Comment by Neules** [comment]
Same here, strip leds doesn't power off when on PC shutdown.

My setup:

[BTF-Lightning W2812B strip led](https://es.aliexpress.com/item/2036819167.html?spm=a2g0s.9042311.0.0.281d63c055GZMO)

PC:
Gigabyte Aorus elite v2 + ryzen 5 5600X
Windows 10 Pro

Pending to install Linux in a second ssd, will report back then.

**Comment by Benik3** [comment]
If you try to shut off the LEDs by doubleclick on Prismatik icon, does it also always turn off?

**Comment by 1MikeKramer** [comment]
I saw this thread and wanted to add my information to it as well.
I built my setup about 2 years ago, I have gone through several versions prismatic and have always had the problem of the lights not turning off when I shut down the computer.
I use the mood lamp static color for times when I am doing productive work and find this mode sticks on during shutdown much more often than the grabbing mode. (Though grabbing mode stays on during shutdown about 30% of the time)
I always have to cycle the lights on and off like you said Benik3 by double clicking on the icon before I exit the software to get the lights to turn off. 
95% of the time the lights cycle normally on and off when clicking the icon.
There is the rare instance where double clicking the icon does nothing, I can switch profiles, but the leds are stuck on. Only by exiting prismatic then re-launching it do I get control of the leds back again, can switch profiles, and cycle them on and off like normal.

Not sure if this information helps at all, But figured since this was a relevant discussion I should share my experience.

**Comment by Neules** [comment]
> If you try to shut off the LEDs by doubleclick on Prismatik icon, does it also always turn off?

Yes, 90% of the time (it fails sometimes but it doesn't worry me).

**Comment by Benik3** [comment]
When turning PC off there could be theoretically problem, that SW is not fast enough to send the off message (everyone is using SSD now ad OS is pretty fast in turning ON/OFF).
The problem with not turning off by doubleclick is probably cause by the Arduino Atmega + WS2812, mentioned here: #223

You can also easily solve the problem with LEDs by simply turning OFF the power to LED strip, which can be done simply by MOSFET/relay connected to Arduino power...

**Comment by Neules** [comment]
I'm a beginner with electronics... I'll look forward for your advice on MOSFET with the arduino, thanks a lot for your answer!

**Comment by Ekristoffe** [comment]
Hello all, I have found this fork for arduino:
https://github.com/t0stiman/adalight_ws2812
It use a system of timeout to shutdown the LED after some time when no data is coming from the serial port.
It could help to give it a try.

**Comment by Benik3** [comment]
I don't see any timeout in the code but yes, this could be an option.

**Comment by Ekristoffe** [comment]
thanks my mistake, I use the same nearly the same code (timeout is 30s for me)

**Comment by Ekristoffe** [comment]
@Infidaelity @1MikeKramer @Benik3 I have found a better program for the arduino base:
https://github.com/dmadison/Adalight-FastLED

I hope this could help a lot like it helped me to have a more stable setup.

**Comment by Benik3** [comment]
If you want more stable setup use ESP8266/ESP32 and Makuna NeopixelBus library, instead of Atmega and FastLED :)
https://github.com/Benik3/Adalight_WS2812_ESP8266_Non-blocking

**Comment by Ekristoffe** [comment]
That can be interesting to test. If I remember right the arduino Nano every I am using could also use another uart port for the LED.
And for the lvl problem maybe using a fast optocopler or a transistor.

**Comment by Benik3** [comment]
But Arduino Nano (Atmega 328P) has only one UART, which is used for communication from Prismatik.  
Yes, you can of course use fast enough level shifter. For WS2812 etc. it's not so problematic, for SPI (APA102, SK9822 etc.) it's worse. But Makuna NeopixelBus has also option to invert the output data, so you can use simply just one transistor as level shifter. I like this solution with diode, it's easy and working nice. I have it now on my TV.

**Comment by Ekristoffe** [comment]
the nano every use the ATMega4809, it seem to have 4 USART ... maybe i need to learn more about this little board.

**Comment by Benik3** [comment]
Interesting, I didn't know there is version with Atmega4809. The right name is Arduino Nano Every.  
It's a nice upgrade, but I don't know if FastLED support using serial as output for LEDs. In Makuna NeopixelBus there is supported "Many Arm based Arduinos", but also not sure if it can use the serial port to output data for LED without disabling interrupts on these platforms.

**Comment by Ekristoffe** [comment]
Anyway I will download your .ino and test it on my hardware.
if it doesn't work what could go wrong XD

**Comment by wintercounter** [comment]
I had the same problem. Turns out it was because of Over-Brightening. Turned it off (to 0) and now LEDs are turning off fine as well. Played a bit with Gamma instead to achieve similar results. It's funny tho, when just my screen turns off, LEDs are turning off fine despite of Over-Brightening. Seems like the software should automatically turn it to 0 upon suspend.

**Comment by Ekristoffe** [comment]
this is because of the controller used. some time it will loose some frames ...
Unfortunately it may also loo the last frame (go black) ...

**Comment by wintercounter** [comment]
That doesn't explain why it works just fine without Over-Brightening.
